safekeeper/src/remove_wal.rs

```diff
@@ -3,18 +3,15 @@ use utils::lsn::Lsn;
 use crate::timeline_manager::StateSnapshot;
 
 /// Get oldest LSN we still need to keep. We hold WAL till it is consumed
-/// by all of 1) pageserver (remote_consistent_lsn) 2) peers 3) s3
-/// offloading.
+/// by 1) pageserver (remote_consistent_lsn) 2) s3 offloading. If some peer
+/// safekeeper misses data it will fetch it from the remote storage.
 /// While it is safe to use inmem values for determining horizon,
 /// we use persistent to make possible normal states less surprising.
 /// All segments covering LSNs before horizon_lsn can be removed.
 pub(crate) fn calc_horizon_lsn(state: &StateSnapshot, extra_horizon_lsn: Option<Lsn>) -> Lsn {
     use std::cmp::min;
 
-    let mut horizon_lsn = min(
-        state.cfile_remote_consistent_lsn,
-        state.cfile_peer_horizon_lsn,
-    );
+    let mut horizon_lsn = state.cfile_remote_consistent_lsn;
```

Comment:
Let's add `commit_lsn` and `flush_lsn` here, otherwise we can observe LSN from the future in case current safekeeper is lagging.

Reply:
done